### ISSUE TITLE
Unsafe context required on an explicit interface implementation with pointers in the interface type parameters despite being marked unsafe

### ISSUE THREAD

**Issue by IS4Code** [dotnet/roslyn#80565]
**Version Used**: .NET 9 SDK

**Steps to Reproduce**:

```cs
interface ITest<T>
{
    void Method();
}

unsafe interface IPointerTest : ITest<void*[]>
{

}

class PointerImpl : IPointerTest
{
    // error CS0214: Pointers and fixed size buffers may only be used in an unsafe context
    unsafe void ITest<void*[]>.Method()
    {

    }
}
```

**Diagnostic Id**:

CS0214: Pointers and fixed size buffers may only be used in an unsafe context

**Expected Behavior**:

The use of `void*` should be okay because the member is marked `unsafe`.

**Actual Behavior**:

An error is reported and the `unsafe` modifier on the member appears to be ignored within the interface type reference. The code works only when the modifier is put on the whole class.
